internal/database/lock.go

```diff
@@ -89,6 +91,36 @@ func makeMultiUnlockFn(ctx context.Context, db *DB, lockIDs []string, expires ti
 	}
 }
 
+func serializationError(err error) bool {
+	// See https://www.postgresql.org/docs/current/errcodes-appendix.html
+	if pgErr, ok := err.(*pgconn.PgError); !ok || pgErr.Code == "40001" {
+		return true
+	}
+	return false
+}
+
+// LockRetry will attempt to acquire a lock using exponential backoff, up to the tryFor time.
+// Delegates to db.Lock().
+func (db *DB) LockRetry(ctx context.Context, lockID string, ttl time.Duration, tryFor time.Duration) (UnlockFn, error) {
```

Comment:
I'm slightly confused. You wrote this nice function `LockRetry`but it doesn't seem to be used anywhere?

Reply:
will do in follow up - wanted to make sure this was agreeable structure for retries.

Reply:
I agree